tutorials/examples/basic_image_array.py

```diff
@@ -0,0 +1,184 @@
+import colorednoise
+import matplotlib._color_data as mcd
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy import misc
+from skimage.measure import block_reduce
+
+from aspire.image import Image
+from aspire.noise import WhiteNoiseEstimator
+from aspire.source import ArrayImageSource
+
+# ------------------------------------------------------------------------------
+# Lets get some basic image data as a numpy array.
+
+# Scipy ships with a portrait of a Procyon (common north american trash panda).
+img_data = misc.face(gray=True)
+
+# Crop to a square
+n_pixels = min(img_data.shape)
+img_data = img_data[0:n_pixels, 0:n_pixels]
+# Block (down)sample the image.
+img_data = block_reduce(img_data, (2, 2))
+
+plt.imshow(img_data, cmap=plt.cm.gray)
+plt.title("Starting Image")
+plt.show()
+
+# Initially, we will add some white noise to the image.
+# Later we will try other types of noise in combination with whitening.
+
+mu = 0.0
+sigma = 256.0
+white = np.random.normal(mu, sigma, img_data.shape)
+
+
+# We'll also compute the spectrum of the original image and white noise sample for later.
+img_data_f = np.abs(np.fft.fftshift(np.fft.fft2(img_data)))
+white_f = np.abs(np.fft.fftshift(np.fft.fft2(white)))
+
+
+# ------------------------------------------------------------------------------
+# Now that we have an example array, we'll begin using the ASPIRE toolkit.
+
+# First we'll make an ASPIRE Image instance out of our data.
+# This is a light wrapper over the numpy array. Many ASPIRE internals
+# are built around an Image classes.
+
+# Construct the Image class by passing it an array of data.
+img = Image(img_data + white)
+
+# "Source" classes are what we use in processing pipelines.
+# They provide a consistent interface to a variety of underlying data sources.
+# In this case, we'll just use our Image to run a small experiment.
+# If you were happy with the experiment design on an array of test data,
+# the source is easily swapped out to something like RelionSource,
+# which might point at a stack of images too large to fit in memory at once.
+# This would be managed behind the scenes for you.
+img_src = ArrayImageSource(img)
+
+# ASPIRE's WhiteNoiseEstimator consumes from a Source
+noise_estimator = WhiteNoiseEstimator(img_src)
```

Comment:
This is not going to do anything since that estimator assumes that the noise is already white. You'll want something like `AnisotropicNoiseEstimator`.

Reply:
Sure, I'll try it.